scripts/signup.js

```diff
@@ -0,0 +1,114 @@
+document.addEventListener("DOMContentLoaded", function () {
+  const emailInput = document.getElementById("email");
+  const nicknameInput = document.getElementById("nickname");
+  const passwordInput = document.getElementById("password");
+  const passwordConfirmInput = document.getElementById("passwordConfirmation");
+  const signupForm = document.querySelector(".auth-form");
+  const signupButton = signupForm.querySelector('button[type="submit"]');
+  const togglePasswordButtons = document.querySelectorAll(".toggle-password");
+
+  emailInput.addEventListener("blur", function () {
+    if (!isNotEmpty(this.value)) {
+      showError(this, "이메일을 입력해주세요.");
+    } else if (!isValidEmail(this.value)) {
+      showError(this, "잘못된 이메일 형식입니다.");
+    } else {
+      removeError(this);
+    }
+    validateForm();
+  });
+
+  nicknameInput.addEventListener("blur", function () {
+    if (!isNotEmpty(this.value)) {
+      showError(this, "닉네임을 입력해주세요.");
+    } else {
+      removeError(this);
+    }
+    validateForm();
+  });
+
+  passwordInput.addEventListener("blur", function () {
+    if (!isNotEmpty(this.value)) {
+      showError(this, "비밀번호를 입력해주세요.");
+    } else if (!isValidPasswordLength(this.value)) {
+      showError(this, "비밀번호를 8자 이상 입력해주세요.");
+    } else {
+      removeError(this);
+    }
+    validatePasswordMatch();
+    validateForm();
+  });
+
+  passwordConfirmInput.addEventListener("blur", function () {
+    validatePasswordMatch();
+    validateForm();
+  });
+
+  function validatePasswordMatch() {
+    if (isNotEmpty(passwordConfirmInput.value)) {
+      if (!isMatching(passwordInput.value, passwordConfirmInput.value)) {
+        showError(passwordConfirmInput, "비밀번호가 일치하지 않습니다.");
+      } else {
+        removeError(passwordConfirmInput);
+      }
+    }
+  }
```

Comment:
💊 제안
if문이 중첩되면 가독성이 떨어지므로 아래처럼 작성하시는 것을 추천드립니다.

```suggestion
  function validatePasswordMatch() {
    if (!isNotEmpty(passwordConfirmInput.value)) return;
    if (!isMatching(passwordInput.value, passwordConfirmInput.value)) {
      showError(passwordConfirmInput, "비밀번호가 일치하지 않습니다.");
    } else {
      removeError(passwordConfirmInput);
    }
  }
```